# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaParent.cpp
# Commit: d63dac1ee4d4
# Full Hash: d63dac1ee4d471defff484dae5a3535e1f8ba072
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-10 22:03:45
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Make it possible to initialize temporary origins without ensuring origin directories; r=dom-storage-reviewers,jari
#   
#   LSNG already uses some QuotaManager APIs to achieve that origin directories are
#   not created if they don't exist during datastore preparation, but the feature
#   is not easy to use and it's also not generalized enough for use in other quota
# ==============================================================================

diff -r 2ab08c49f2b1 -r d63dac1ee4d4 dom/quota/QuotaParent.cpp
--- a/dom/quota/QuotaParent.cpp	Thu Oct 10 12:22:54 2024 +0000
+++ b/dom/quota/QuotaParent.cpp	Thu Oct 10 12:22:55 2024 +0000
@@ -440,7 +440,7 @@
 
 mozilla::ipc::IPCResult Quota::RecvInitializeTemporaryOrigin(
     const PersistenceType& aPersistenceType,
-    const PrincipalInfo& aPrincipalInfo,
+    const PrincipalInfo& aPrincipalInfo, const bool& aCreateIfNonExistent,
     InitializeTemporaryOriginResolver&& aResolve) {
   AssertIsOnBackgroundThread();
 
@@ -459,7 +459,9 @@
                 QuotaManager::GetOrCreate(),
                 ResolveBoolResponseAndReturn(aResolve));
 
-  quotaManager->InitializeTemporaryOrigin(aPersistenceType, aPrincipalInfo)
+  quotaManager
+      ->InitializeTemporaryOrigin(aPersistenceType, aPrincipalInfo,
+                                  aCreateIfNonExistent)
       ->Then(GetCurrentSerialEventTarget(), __func__,
              BoolPromiseResolveOrRejectCallback(this, std::move(aResolve)));
 